lib/ecto/repo/preloader.ex

```diff
@@ -170,15 +170,23 @@ defmodule Ecto.Repo.Preloader do
         acc
       struct, {fetch_ids, loaded_ids, loaded_structs} ->
         assert_struct!(module, struct)
-        %{^owner_key => id, ^field => value} = struct
+        %{^field => value} = struct
+        ids = Enum.map(owner_key, fn key -> 
+          %{^key => id} = struct
+          id
+        end)
         loaded? = Ecto.assoc_loaded?(value) and not force?
 
-        if loaded? and is_nil(id) and not Ecto.Changeset.Relation.empty?(assoc, value) do
+        if loaded? and Enum.any?(ids, &is_nil/1) and not Ecto.Changeset.Relation.empty?(assoc, value) do
+          {key_noun, key_verb_present, key_verb_past, key_list} = case owner_key do
+            [single_key] -> {"key", "is",  "was", single_key}
+            many_keys -> {"keys", "are", "were", "[#{Enum.join(many_keys, ", ")}]"}
+          end
```

Comment:
I think we can simplify it and leave only the plural form. Something like:

    its associations keys (`foo_id`) are nil.